cli_tools/gce_vm_image_export/exporter/exporter.go

```diff
@@ -226,11 +225,22 @@ func Run(clientID string, destinationURI string, sourceImage string, sourceDiskS
 	return w, nil
 }
 
-func validateImageExists(computeClient daisyCompute.Client, project string, sourceImage string) (err error) {
-	_, err = computeClient.GetImage(project, sourceImage)
+// validateImageExists checks whether imageName exists in the specified project.
+//
+// This validates when imageName is a valid image name, and skips validation if
+// the imageName is a URI, or something that's not recognized as an image.
+// The simplistic validation avoids false negatives; Daisy has robust logic for
+// interpreting the various permutations of specifying an image and project,
+// and we don't want to copy that here, since this is a convenience method to create
+// user-friendly messages.
+func validateImageExists(computeClient daisyCompute.Client, project string, imageName string) (err error) {
+	if err := validation.ValidateImageName(imageName); err != nil {
```

Comment:
> What about extracting project/imageName from image URL (if that's what's provided) and validating based on those?

@zoran15: Would you be alright going forward with the current version, and leaving parsing as a potential future enhancement?  

I'm concerned about the number number of URI formats that are possible, and the amount of work required to get all of the correct:

- Image using family, eg: `projects/debian-cloud/global/images/family/debian-9`
- Image without family but different project, `projects/debian-cloud/global/images/debian-9-stretch-vYYYYMMDD`
- Potential leading API qualifier, that could be incorrect: **http://www.googleapis.com/compute/v1**
- Partial URI that includes the name but not project: `global/images/image-name`

The cost of failure is blocking an export, while the benefit is a better error message. Furthermore, I wanted to avoid duplicating this parsing logic in our codebase, since Daisy already has this implemented.

Reply:
Sure, let's keep it simple for now. We should create a tracking bug for expanding this functionality.